### PR TITLE
metrics: add worker thread id

### PR DESCRIPTION
**Pull request by surban**
This allows querying the thread id of each worker thread.

Additional information, such as native thread id, can be collected using `runtime::Builder::on_thread_start()` and correlated using the thread id.

## Motivation

See #6353 and discussion in #6370.

## Solution

Provides thread ids of workers, so that a stuck worker can be correlated to a thread.

### CODE REVIEW

File: tokio/tests/rt_unstable_metrics.rs
Comment (Darksonn) [inline tokio/tests/rt_unstable_metrics.rs:1]: I think this is a good idea, but there are a bunch of cases where the thread id of a worker can change:

* The current thread runtime can move as you call `block_on` from different threads.
* The multi thread runtime can have the worker threads move when `block_in_place` is used.

I would like to make sure that these cases are handled properly. Would you be able to add some tests for them?

File: tokio/tests/rt_unstable_metrics.rs [hunk 1]
```diff
@@ -150,6 +151,28 @@ fn remote_schedule_count() {
     assert_eq!(1, rt.metrics().remote_schedule_count());
 }
 
+#[test]
+fn worker_thread_id() {
```
Comment (Darksonn) [inline tokio/tests/rt_unstable_metrics.rs:155]: (No need to put everything you test related to thread ids in a single test. You can make more than one.)

File: tokio/tests/rt_unstable_metrics.rs
Comment (surban) [reply tokio/tests/rt_unstable_metrics.rs:1]: I added the necessary tests.

File: tokio/src/runtime/metrics/runtime.rs [hunk 2]
```diff
@@ -127,6 +128,49 @@ impl RuntimeMetrics {
...
         }
 
+        /// Returns the thread id of the given worker thread.
+        ///
+        /// The returned value is `None` if the worker thread has not yet finished
+        /// starting up.
+        ///
+        /// If additional information about the thread, such as its native id, are
+        /// required, those can be collected in [`on_thread_start`] and correlated
+        /// using the thread id.
```
Comment (Darksonn) [inline tokio/src/runtime/metrics/runtime.rs:138]: This is enough for your purposes?

**Review by Darksonn — APPROVED** [review]
Thanks. This looks good to me.

File: tokio/tests/rt_unstable_metrics.rs
Comment (Darksonn) [reply tokio/tests/rt_unstable_metrics.rs:1]: Thanks. The tests make it easy for me to believe that you are updating the stored thread id in all of the right places.

File: tokio/src/runtime/metrics/runtime.rs (hunk 2 above)
Comment (surban) [reply tokio/src/runtime/metrics/runtime.rs:138]: Yes, for now we just want to send SIGABRT to the hanging thread and for that we collect the `pthread_t` id of each worker thread in `on_thread_start`.

**Pull request merged**